repository/index.go

```diff
@@ -54,6 +55,27 @@ func (idx *Index) Store(t pack.BlobType, id backend.ID, pack *backend.ID, offset
 	idx.store(t, id, pack, offset, length, false)
 }
 
+// Adds a preliminary index entry for a blob that is about to be saved. The entry must be updated using Store once
+// the the blob has been written to a pack. Adding an preliminary index fails if there's an existing entry associated
+// with the same id. Due to locking the index this will ensure only one goroutine processes the the same blob
+// concurrently.
+func (idx *Index) StoreInProgress(t pack.BlobType, id backend.ID) error {
+	// Note: Cannot use idx.Has(...) or idx.Store(...) as these would lock the index again and Go mutexes seem not to be
```

Comment:
Comment line is too wide, please wrap at 80 chars. Is this comment really necessary?